File: euler_004.py
from time import time
from itertools import product

def is_palindrome(num):
    return str(num) == str(num)[::-1]

start = time()

m = max([d[0]*d[1] for d in product(range(100,1000),repeat=2) if is_palindrome(d[0]*d[1])])
print("Largest palindrome: " + str(m))

stop = time()
print("Time: " + str(stop-start))

File: euler_004.js
const start = Date.now();

const isPalindrome = (num) => {
    const asString = num.toString();
    for (let i = 0;i <= Math.floor(asString.length / 2);i++) {
        if (asString[i] !== asString[asString.length - 1 - i]) {
            return false;
        }
    }
    return true;
}

let largestPalindrome = 1;
const threeDigitNumbers = [...Array(1000).keys()].slice(100);

for (const num1 of threeDigitNumbers) {
    for (const num2 of threeDigitNumbers) {
        if (num1 >= num2 && isPalindrome(num1 * num2) && largestPalindrome < num1 * num2) {
            largestPalindrome = num1 * num2;
        }
    }
}

const end = Date.now();

console.log("Execute time: ", end - start, " Largest palindrome is: ", largestPalindrome);